### PR TITLE
z3: add 4.11

### PR DESCRIPTION
**Pull request by macalinao**
###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [x] (Module addition) Added a release notes entry if adding a new NixOS module
  - [x] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -34634,9 +34634,10 @@ with pkgs;
 
 
   inherit (callPackages ../applications/science/logic/z3 { python = python2; })
+    z3_4_11
     z3_4_8
     z3_4_7;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:34639]:
```suggestion
```
Can we drop 4.7?

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -34634,9 +34634,10 @@ with pkgs;
...
-  z3 = z3_4_8;
+  z3 = z3_4_11;
   z3_4_4_0 = callPackage ../applications/science/logic/z3/4.4.0.nix {
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:34641]: or 4.4.0?

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (macalinao) [reply pkgs/top-level/all-packages.nix:34639]: What if someone is using it?

I'm not sure what the standard is here but I'll do this if it's what I should do.

Comment (risicle) [reply pkgs/top-level/all-packages.nix:34639]: If you _do_ choose to remove it, add it to `aliases.nix`.

**Comment by risicle** [comment]
On macos 10.15, `haskellPackages.z3` fails to build with:

```
...
rsp file was: "dist/build/Z3/Base/hsc2hscall66303-0.rsp"
command was: /nix/store/md0r2kkc0wg2gk18aaim529krci9kv3w-clang-wrapper-11.1.0/bin/clang -c dist/build/Z3/Base/C_hsc_make.c -o dist/build/Z3/Base/C_hsc_make.o --target=x86_64-apple-darwin --target=x86_64-apple-darwin --target=x86_64-apple-darwin -D__GLASGOW_HASKELL__=900 -Ddarwin_BUILD_OS=1 -Dx86_64_BUILD_ARCH=1 -Ddarwin_HOST_OS=1
error: C.hsc:97:16: error: use of undeclared identifier 'Z3_TRUE'; did you mean 'Z3_L_TRUE'?
    hsc_const (Z3_TRUE);
               ^~~~~~~
               Z3_L_TRUE
...
```

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (ttuegel) [reply pkgs/top-level/all-packages.nix:34639]: Please do _not_ alias it. Different versions of Z3 are completely incompatible.

**Comment by macalinao** [comment]
@risicle I changed the default z3 version back to 4.8 to "fix" this

File: pkgs/top-level/all-packages.nix (hunk 1 above)
Comment (risicle) [reply pkgs/top-level/all-packages.nix:34639]: `aliases.nix` is largely used to "alias" removed packages with `throw` statements providing guidance.

**Comment by macalinao** [comment]
sorry for pinging you guys repeatedly-- i don't know how to use GitHub

**Review by risicle — APPROVED** [review]
Builds for me, macos 10.15 & nixos x86_64

**Comment by macalinao** [comment]
Builds for me on macos 12 arm64

**Comment by macalinao** [comment]
any idea what it would take to get this merged? No rush-- just curious if I've missed anything. Thanks!

**Comment by macalinao** [comment]
Bump-- this doesn't modify anything so i feel like it should be relatively easy to merge in

**Review by risicle — APPROVED** [review]

**Pull request merged**